src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ZERO = 0;
+    private static final int AUXILIARY_ONE = 1;
```

Comment:
wasteful constants
Do you think looking at 'ZERO' someone will better understand your code than looking at '0'?

Reply:
not fixed - delete these constants